pyccel/ast/core.py

```diff
@@ -2919,6 +2927,51 @@ def funcdef(self):
     def interface(self):
         return self._interface
 
+    @property
+    def func_name(self):
+        return self._func_name
+
+    @property
+    def interface_name(self):
+        return self._interface_name
+
+class DottedFunctionCall(FunctionCall):
```

Comment:
if `DottedFunctionCall` inherits from `FunctionCall` then `DottedVariable` should do the same thing and inherit from `Variable`

Reply:
Yes, this is being done in PR #637. But that PR was getting excessively large so I extracted this part which didn't seem relevant. Would you prefer that I moved that change to this PR?

Reply:
it's up to you , if it's in another PR I guess it's okay, but probably grouping them in one PR would be good also.

Reply:
Ok. Then I will use this PR for all the Dotted improvements